tools/evaluation.py

```diff
@@ -149,12 +149,20 @@ def main():
         else:
             basic_table_info['num_samples'] = -1
             # build the dataloader
-            if cfg.data.get('test', None):
+            if cfg.data.get('test', None) and cfg.data.test.get(
+                    'imgs_root', None):
                 dataset = build_dataset(cfg.data.test)
-            elif cfg.data.get('val', None):
+            elif cfg.data.get('val', None) and cfg.data.val.get(
+                    'imgs_root', None):
                 dataset = build_dataset(cfg.data.val)
-            else:
+            elif cfg.data.get('train', None) and \
```

Comment:
Here, I may suggest that we can just use `cfg.data.get('train', None)`. Because we assume that the `train` part should work well once it has been added.